demo_scripts/AKSDeployment/README.md

```diff
@@ -0,0 +1,314 @@
+Welcome to this tutorial where we will take you step by step in creating an AKS Application with a custom domain that is secured via https. This tutorial assumes you are logged into Azure CLI already and have selected a subscription to use with the CLI. It also assumes that you have helm installed (Instructions ca be found here https://helm.sh/docs/intro/install/). If you have not done this already. Press b and hit ctl c to exit the program.
+
+To Login to Az CLI and select a subscription 
+'az login' followed by 'az account list --output table' and 'az account set --subscription "name of subscription to use"'
+
+To Install Az CLI
+If you need to install Azure CLI run the following command - curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+
+
+Assuming the pre requisites are met press enter to proceed
+
+## Define Command Line Variables 
+Most of these variables should be set to a smart default. However, if you want to change them
+press b and run the command export VARIABLE_NAME="new variable value"
+
+```
+echo $RESOURCE_GROUP_NAME
+echo $RESOURCE_LOCATION
+echo $AKS_CLUSTER_NAME
+echo $PUBLIC_IP_NAME
+echo $VNET_NAME
+echo $SUBNET_NAME
+echo $APPLICATION_GATEWAY_NAME
+```
+
+For the following variables, unless you manually added them in the env.json you will be asked to provide an input
+
+The custom domain must be unique and fit the pattern: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$
+For example mycooldomain 
+
+Do not add any capitalization or .com
+```
+echo $CUSTOM_DOMAIN_NAME
+```
+
+For the email address any enter a valid email. I.e sarajane@gmail.com
+```
+echo $SSL_EMAIL_ADDRESS
+```
+
+## Create A Resource Group
+The first step is to create a resource group. You can do this with the following command
+
+'az group create --name $RESOURCE_GROUP_NAME --location $RESOURCE_LOCATION'
+
+```
+az group create --name $RESOURCE_GROUP_NAME --location $RESOURCE_LOCATION
+```
+
+## Create an AKS Cluster 
+The next step is to create an AKS Cluster. This can be done with the following command - 
+
+az aks create --resource-group $RESOURCE_GROUP_NAME --name $AKS_CLUSTER_NAME --node-count 1 --enable-addons monitoring --generate-ssh-keys
+
+This will take a few minutes
+```
+az aks create --resource-group $RESOURCE_GROUP_NAME --name $AKS_CLUSTER_NAME --node-count 1 --enable-addons monitoring --generate-ssh-keys
+```
+
+## Install az aks CLI
+To manage a Kubernetes cluster, use the Kubernetes command-line client, kubectl. kubectl is already installed if you use Azure Cloud Shell.
+
+Install kubectl locally using the az aks install-cli command:
+
+az aks install-cli
+
+```
+az aks install-cli
```

Comment:
If `kubectl` is already installed this will fail with an error. We should test if it is present and only execute if it is not.

Reply:
I definitely agree that would be a better user experience. My thought in the PoC is that the program continues to execute with no harm done if it is already installed.

Reply:
Sure but it's super simple to do, so what's stopping us?
```
if ! [ -x "$(command -v kubectl)" ]; then az aks install-cli; fi
```

Reply:
Haha my knowledge of Bash was stopping me ;) I will make the change right now!